pkg/sql/distsqlrun/windower.go

```diff
@@ -617,7 +618,19 @@ func (w *windower) processPartition(
 				if err != nil {
 					return err
 				}
+				if prevRes == nil || prevRes != res {
+					// We don't want to double count the same memory, and since the same
+					// memory can only be reused contiguously as res, comparing against
+					// result of the previous row is sufficient.
+					// We have already accounted for the size of a nil datum prior to
+					// allocating the slice for window values, so we need to keep that in
+					// mind.
+					if err := w.growMemAccount(&w.acc, int64(res.Size())-sizeOfDatum); err != nil {
```

Comment:
tip: over-accounting by a fix, small amount is not a problem. It does not impair the correctness of the code, and may make it easier to read/understand.

Reply:
I agree in general, but here this amount is dependent on the number of the rows in the partition of a window function, so the over-accounting will grow linearly (which still might be not a big deal, but I don't think here the pattern is the code is complicated enough to justify not doing the more precise accounting).